src/game_battler.cpp

```diff
@@ -266,17 +266,21 @@ bool Game_Battler::UseSkill(int skill_id, const Game_Battler* source) {
 		}
 
 		// Calculate effect:
-		// FIXME: What about negative attributes? Does it do damage here? Can it kill?
 		auto effect = Algo::CalcSkillEffect(*source, *this, *skill, true);
 
+		// Negative attributes do damage but cannot kill
+		bool negative_effect = false;
+		if (effect < 0) {
+			negative_effect = true;
+			effect = -effect;
+		}
+
 		// Cure states
 		for (int i = 0; i < (int)skill->state_effects.size(); i++) {
 			if (skill->state_effects[i]) {
 				if (skill->reverse_state_effect) {
 					was_used |= !HasState(lcf::Data::states[i].ID);
-					// FIXME: This logic always called from menu, so we set false
-					// to allow_battle_states
-					AddState(lcf::Data::states[i].ID, false);
+					AddState(lcf::Data::states[i].ID, true);
```

Comment:
Not sure how we missed this last time. You're right that this case allows you to inflict battle only states, even from the menu.